src/sp/it/pl/plugin/LibraryWatcher.kt

```diff
@@ -0,0 +1,143 @@
+package sp.it.pl.plugin
+
+import mu.KLogging
+import org.reactfx.Subscription
+import sp.it.pl.audio.SimpleItem
+import sp.it.pl.audio.tagging.MetadataReader
+import sp.it.pl.main.APP
+import sp.it.pl.service.notif.Notifier
+import sp.it.pl.util.async.executor.EventReducer
+import sp.it.pl.util.async.runFX
+import sp.it.pl.util.async.runNew
+import sp.it.pl.util.collections.materialize
+import sp.it.pl.util.conf.EditMode
+import sp.it.pl.util.conf.IsConfig
+import sp.it.pl.util.conf.c
+import sp.it.pl.util.conf.cList
+import sp.it.pl.util.conf.cr
+import sp.it.pl.util.conf.cv
+import sp.it.pl.util.conf.only
+import sp.it.pl.util.conf.readOnlyUnless
+import sp.it.pl.util.file.FileMonitor
+import sp.it.pl.util.file.isChildOf
+import sp.it.pl.util.reactive.Subscribed
+import sp.it.pl.util.reactive.onItemAdded
+import sp.it.pl.util.reactive.onItemRemoved
+import sp.it.pl.util.reactive.sync
+import sp.it.pl.util.system.Os
+import sp.it.pl.util.text.plural
+import sp.it.pl.util.validation.Constraint
+import java.io.File
+import java.nio.file.StandardWatchEventKinds.ENTRY_CREATE
+import java.nio.file.StandardWatchEventKinds.ENTRY_DELETE
+import java.util.concurrent.atomic.AtomicLong
+
+class LibraryWatcher: PluginBase("Library Watcher", false) {
+
+    @IsConfig(name = "Location", info = "Locations to find audio.")
+    private val sourceDirs by cList<File>().only(Constraint.FileActor.DIRECTORY)
+    private val sourceDirsChangeHandler = Subscribed {
+        sourceDirs.forEach { handleLocationAdded(it) }
+        Subscription.multi(
+                sourceDirs.onItemAdded { handleLocationAdded(it) },
+                sourceDirs.onItemRemoved { handleLocationRemoved(it) }
+        )
+    }
+
+    @IsConfig(name = "Directory monitoring supported", info = "On some system, this feature may be unsupported", editable = EditMode.NONE)
+    val fileMonitoringSupported by c(Os.WINDOWS.isCurrent)
```

Comment:
Maybe just call it `monitoringSupported` and `monitoringEnabled`
Or libraryMonitoring, but fileMonitoring is a bit vague

Reply:
These variables need to share nomenclature:
- fileMonitoringSupported
- fileMonitoringEnabled
- fileMonitors
- fileMonitoring
Suggested 'monitoring' or 'libraryMonitoring' is more vague than 'fileMonitoring'.

Reply:
Yes. I would propose folderMonitoring or directoryMonitoring then, because fileMonitoring sounds like it monitors individual files, but it monitors folders.

Reply:
I would stil largue that the monitored objects are files not directories thought. Directories only have the monitors registered on them and that is only for naive recursive monitoring implementation. Done properly, there is only one monitor (per root directory), registered on the path denoting the directory. The important part is that we get change events - for FILES. Hence the target of monitoring are the files, not directories, those are the source of the monitoring.

But I'm fine both ways. Shall I use 'dirMonitoring' then?

Reply:
yes

Reply:
Fixed with 'dir' prefix